### ISSUE TITLE
[feature]: Create Profile Modal

### ISSUE THREAD

**Issue by gbowne1** [gbowne1/reactsocialnetwork#199]
### Is your feature request related to a problem? Please describe.

While going back through the login process in a new install on a VM..  I noticed that in the new user process, there is no Create Profile workflow right after registering.  The Registration process should take you right to some sort of page or modal or something that will allow the user to create a profile.  This currently does not exist, and there is currently no way to access the users profile.

We do have a Profile.jsx and a UserProfile.jsx component, but neither one is complete nor houses the functionailty or the UI of creating a Profile.  One of these could be converted to use as a form for entering the users details.

### Describe the solution you would like

- A modal with a form 
- Use one of the 1318.8px x 700px panels to house a form with the necessary bits to create a user profile
- A z-index page with a hidden or transparent background that has a modal or some other panel

This is also an issue #52 associated with this.

### Describe the alternatives you have tried

Currently we have no alternative method either.

**Comment by promise-dash** [comment]
Hey, I would like to solve this issue.
Could you please assign me this issue?

Could you please elaborate that which basic information you would like to collect from the user in order to create its profile?
Like what fields you want on the modal.

**Comment by gbowne1** [comment]
@promise-dash 

My initial thoughts are:

Profile picture: Allow users to upload a profile picture or select one from their device's photo library.  Can also opt for the user to use an avatar instead.

Name: Ask users to enter their full name

Also a display name for their profile.

Username: Allow users to choose a unique username that will be displayed in their profile URL and in mentions.

Bio: Allow users to enter a short description of themselves or their interests.

Location: Allow users to enter their location or choose from a list of predefined locations.

Website: Allow users to enter a link to their personal website or social media profiles.

Interests: Allow users to select their interests from a predefined list or enter their own.

Privacy settings: Allow users to choose who can view their profile and posts (e.g., public, friends-only, private).

Notification settings: Allow users to choose how they receive notifications (e.g., email, push notifications) and which types of notifications they want to receive (e.g., new followers, likes, comments).

Security settings: Allow users to change their password, enable two-factor authentication, and view their login history.

**Comment by promise-dash** [comment]
@gbowne1 Im trying to access "/user-profile" route but it's showing a 404 page.
What should I do?

**Comment by gbowne1** [comment]
There should only be a route for /profile at this point.  

I don't think @manuel12 @pawel975 or I have ever decided on the route for a page that allows you to create a profile.

I actually would propose it could be /profile/create/

That particular route /user-profile 404's because there is no route or component to render

**Comment by promise-dash** [comment]
Actually, m trying to access "localhost:3000/profile" but it's redirecting me to "localhost:3000/login".

**Comment by gbowne1** [comment]
Have you logged in? @promise-dash ?

**Comment by promise-dash** [comment]
yes I was logged in . Ok let me clone the project again

**Comment by gbowne1** [comment]
@promise-dash I am able to commit and push sync just fine.  I have no idea what might be causing your errors, unless youre trying to force push.

**Comment by pawel975** [comment]
> @gbowne1 I m not able to push the changes. It's showing permission denied. Could you please look into this matter?

@promise-dash Did you push from your forked repository? Access is denied when you try to push from cloned repository.

**Comment by gbowne1** [comment]
Yeah, I was going to say that clones have no access.  Only forks.  Some times there are issues when the local fork has not been synced to main too.  I understood that clones only work if you are a owner or maintainer.  I am able to commit, push, pull, etc from a clone (my Dev VM) while my dev PC has only a fork and I am able to use git commit, push, etc from either one.

**Comment by gbowne1** [comment]
Closing as merged
